odl/solvers/advanced/proximal_operators.py

```diff
@@ -103,14 +107,118 @@ def make_diag(step_size):
     return make_diag
 
 
+def proximal_convexconjugate(proximal):
+    """ Calculate the proximal of the dual using Moreau decomposition
+
+    The Moreau identity states that for any convex function ``F`` with
+    convex conjugate ``F^*``, the proximals satisfy
+
+        prox[a * F^*](x) + a * prox[F / a](x / a) = x
+
+    where ``a`` is a scalar step size. Using this, we find the proximal of the
+    convex conjugate
+
+        prox[a * F^*](x) = x - a * prox[F / a](x / a)
+
+    Note that since ``(F^*)^* = F``, this can be used to get the primal from
+    the dual.
+
+    Parameters
+    ----------
+    proximal : `callable`
```

Comment:
The parameter name suggests that this is already the operator, while it's only the factory. `prox_factory` instead?